plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -494,100 +493,112 @@ public void testMySqlDatetimeType(ZoneId sessionZone)
+                .addRoundTrip("datetime(3)", "TIMESTAMP '1970-01-01 00:00:01.123'", createTimestampType(3), "TIMESTAMP '1970-01-01 00:00:01.123'")
+                .addRoundTrip("datetime(4)", "TIMESTAMP '1970-01-01 00:00:01.1234'", createTimestampType(4), "TIMESTAMP '1970-01-01 00:00:01.1234'")
+                .addRoundTrip("datetime(5)", "TIMESTAMP '1970-01-01 00:00:01.12345'", createTimestampType(5), "TIMESTAMP '1970-01-01 00:00:01.12345'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.123456'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123456'")
 
                 // test rounding for precisions too high for MySQL
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234560'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.12345649999'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234565'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234569'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234560'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123456'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.12345649999'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123456'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234565'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123457'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1970-01-01 00:00:01.1234569'", createTimestampType(6), "TIMESTAMP '1970-01-01 00:00:01.123457'")
 
                 .execute(getQueryRunner(), session, mysqlCreateAndInsert("tpch.test_datetime"));
 
         SqlDataTypeTest.create()
-                // TODO support higher precision timestamps (https://github.com/trinodb/trino/issues/6910)
+                // Mysql supports precision up to 6
                 // before epoch with second fraction
-                //.addRoundTrip("timestamp(7)", "TIMESTAMP '1969-12-31 23:59:59.1230000'", createTimestampType(7), "TIMESTAMP '1969-12-31 23:59:59.1230000'")
-                //.addRoundTrip("timestamp(7)", "TIMESTAMP '1969-12-31 23:59:59.1234567'", createTimestampType(7), "TIMESTAMP '1969-12-31 23:59:59.1234567'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.1230001'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.123000'")
+                .addRoundTrip("datetime(6)", "TIMESTAMP '1969-12-31 23:59:59.1234561'", createTimestampType(6), "TIMESTAMP '1969-12-31 23:59:59.123456'")
```

Comment:
This got uncommented, but also changed from timestamp(7) to 6. any particular reason?

Reply:
It's impossible to roundTrip inputTypes of `timestamp(x)` or `datetime(x)`  for `x>6` with `mysqlCreateAndInsert`
Those tests can be move before L584 as `negative epoch - round down`

Reply:
In this case the original commented code was "incorrect" because it was copied from SQL Server (or something else) I believe. So the new test is correct. It just needs to test `before epoch - with second fraction`. Regardless of rounding etc.